core/src/utils/tap-click.ts

```diff
@@ -169,7 +170,8 @@ const getActivatableTarget = (ev: any): any => {
     const path = ev.composedPath() as HTMLElement[];
     for (let i = 0; i < path.length - 2; i++) {
       const el = path[i];
-      if (el?.classList.contains('ion-activatable')) {
+      // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
+      if (el.classList && el.classList.contains('ion-activatable')) {
```

Comment:
We don't need the optional check here anymore: https://developer.mozilla.org/en-US/docs/Web/API/Element/classList#browser_compatibility

The optional check was for IE support. All browsers that Ionic supports also support classList,

Reply:
Discovered why this check was added. The composed path includes shadow dom elements, which don't have `classList` on their prototype. Likely `instance of ShadowRoot` did not exist back when that check was added.

Updated the logic to account for this.

Reply:
Great catch!